consensus/istanbul/core/commit.go

```diff
@@ -40,21 +41,41 @@ func (c *core) generateCommittedSeal(sub *istanbul.Subject) (blscrypto.Serialize
 	return committedSeal, nil
 }
 
-func (c *core) generateEpochValidatorSetData(blockNumber uint64, newValSet istanbul.ValidatorSet) ([]byte, error) {
+// Generates serialized epoch data for use in the Plumo SNARK circuit.
+// Block number and hash may be information for a pending block.
+func (c *core) generateEpochValidatorSetData(blockNumber uint64, blockHash common.Hash, newValSet istanbul.ValidatorSet) ([]byte, error) {
 	if !istanbul.IsLastBlockOfEpoch(blockNumber, c.config.Epoch) {
 		return nil, errNotLastBlockInEpoch
 	}
+
+	// Serialize the public keys for the validators in the validator set.
 	blsPubKeys := []blscrypto.SerializedPublicKey{}
 	for _, v := range newValSet.List() {
 		blsPubKeys = append(blsPubKeys, v.BLSPublicKey())
 	}
+
 	maxNonSigners := uint32(newValSet.Size() - newValSet.MinQuorumSize())
-	epochData, err := blscrypto.EncodeEpochSnarkData(blsPubKeys, maxNonSigners, uint16(istanbul.GetEpochNumber(blockNumber, c.config.Epoch)))
-	if err != nil {
-		return nil, err
+
+	// TODO(victor): Replace false with a check that the hard fork is inactive.
```

Comment:
Actually need to do this once we have a HF mechanism defined. Otherwise as long as the randomness isn't changing mid sequence (which I don't think it is), it's good from a consensus pov.

Reply:
That's my intention. I figure we should get the code checked in now because it seems unclear when the hard-fork solution will be ready. And from the randomness perspective, this is operating on the proposal hash, which means the block is finalized, so unless I misunderstand something, it is guaranteed not to change.

Reply:
One thing if we check this in now, shouldn't it be `true` b/c we don't want to yet activate this hard fork.

Reply:
Oh 😅 Yes!